### ISSUE TITLE
body_reader.js#createBodyWriter() may corrupt on multi-bytes character

### ISSUE THREAD

**Issue by soplwang** [restify/node-restify#791]
Multi-bytes character like Chinese use two or three or four bytes to encode. When we post request with large body, underly TCP may transfer it over multiple packets.

When TCP split packet in the middle of a multi-bytes character, following code may broken and corrupt the output:
```javascript
//@@ -27,9 +27,9 @@ function createBodyWriter(req) {
30        req.body = '';
31        return (function (chunk) {
32            req.body += chunk.toString('utf8');
33        });
```

Fix:
Use `Buffer` and concat them, then on request done convert at once may fix this issue.

**Comment by soplwang** [comment]
dup with #774 #775

**Comment by DonutEspresso** [comment]
Fixed via #775
